### PR TITLE
uhd: wrap get_usrp_info to compatible return type

### LINKED ISSUE #3608: uhd: pybind cannot convert get_usrp_info to python type
If one creates a uhd source (or sink) and calls `get_usrp_info` on it, the following error is thrown:
```python
>>> d = uhd.usrp_source(device_args, stream_args)
>>> info = d.get_usrp_info()
TypeError: Unable to convert function return value to a Python type! The signature was
	(self: gnuradio.uhd.uhd_python.usrp_source, chan: int = 0) -> uhd::dict<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >

Did you forget to `#include <pybind11/stl.h>`? Or <pybind11/complex.h>,
<pybind11/functional.h>, <pybind11/chrono.h>, etc. Some automatic
conversions are optional and require extra headers to be included
when compiling your pybind11 module.
```

### PR DESCRIPTION
**Pull request by mormj**
Copied the wrapping mechanism into a lambda from pyuhd for these functions

Fixes #3608

### CODE REVIEW

File: gr-uhd/python/uhd/bindings/usrp_sink_python.cc [hunk 1]
```diff
@@ -57,10 +57,13 @@ void bind_usrp_sink(py::module& m)
...
-             &usrp_sink::get_usrp_info,
-             py::arg("chan") = 0,
-             D(usrp_sink, get_usrp_info))
+        .def(
+            "get_usrp_info",
+            [](usrp_sink& self, const size_t chan = 0) {
+                return static_cast<std::map<std::string, std::string>>(
+                    self.get_usrp_info(chan));
+            },
+            py::arg("chan") = 0)
```
Comment (jdemel) [inline gr-uhd/python/uhd/bindings/usrp_sink_python.cc:66]: Except for `usrp_sink` this is exactly the same code as in `usrp_source`.
Can you move this to `usrp_block`?

Comment (mormj) [reply gr-uhd/python/uhd/bindings/usrp_sink_python.cc]: Good point - no reason to have duplicate code.  Second commit is added to address this.

**Comment by mormj** [comment]
Struggling to get CI to pass on this one (it builds on my machine lol).  Perhaps there is a different version of pybind11 between the buildbot and mine.  more investigation.

**Comment by mormj** [comment]
Unfortunately I haven't been able to get this to compile on the docker used in CI - perhaps it is the difference in gcc.  Will have to investigate how to do this with gcc7 or whatever other tools are in the ci containers

File: gr-uhd/lib/usrp_block_impl.cc [hunk 2]
```diff
@@ -741,3 +741,9 @@ void usrp_block_impl::_cmd_handler_dspfreq(const pmt::pmt_t& dspfreq,
 
     _update_curr_tune_req(new_tune_request, chan);
 }
+
+::uhd::dict<std::string, std::string> usrp_block_impl::get_usrp_info(size_t chan)
+{
+    chan = _stream_args.channels[chan];
+    return _dev->get_usrp_rx_info(chan);
```
Comment (ncorgan) [inline gr-uhd/lib/usrp_block_impl.cc:748]: The `multi_usrp` class has both `get_usrp_tx_info` and `get_usrp_rx_info` functions, with some different information between each. The separate implementations in `usrp_source` and `usrp_sink` were intentional for this reason. Consolidating these into `usrp_block` removes getting TX-specific info.

Comment (jdemel) [reply gr-uhd/lib/usrp_block_impl.cc:748]: That's a very good reason to keep them separate. But also, it's a very good reason to update these function names.
I suggest to add `get_usrp_rx_info` to `usrp_source` and `get_usrp_tx_info` to `usrp_sink`. Further, I suggest to keep `get_usrp_info` in the base class, deprecate it and log a warning everytime it is called. Alternatively, we can remove it. But I doubt this approach receives approval without deprecation for at least one release cycle.
Anyways, I'd consider it a bug that the function does not fully describe what it does and needs to be fixed.

Comment (ncorgan) [reply gr-uhd/lib/usrp_block_impl.cc:748]: The `get_usrp_*x_info` solution, with deprecation of the old function name, sounds like a good plan.

Comment (mbr0wn) [reply gr-uhd/lib/usrp_block_impl.cc:748]: Every other function in these blocks has the same name for the sink and source (like set gain, set freq) and is declared in usrp_block, and then defined in the sink or source. I don't see a necessity to change that here.

**Comment by mbr0wn** [comment]
I'm closing this in favour of https://github.com/gnuradio/gnuradio/pull/3691, which compiles and doesn't have the drawbacks mentioned here. @jdemel, @ncorgan, @mormj I'd be very happy if you could take another look there.

**Pull request closed without merging**